agent/src/com/thoughtworks/go/agent/AgentController.java

```diff
@@ -62,13 +64,17 @@
     private SubprocessLogger subprocessLogger;
     private final SystemEnvironment systemEnvironment;
     private AgentUpgradeService agentUpgradeService;
+    private PackageAsRepositoryExtension packageAsRepositoryExtension;
+    private SCMExtension scmExtension;
     private TaskExtension taskExtension;
 
     @Autowired
     public AgentController(BuildRepositoryRemote server, GoArtifactsManipulator manipulator, SslInfrastructureService sslInfrastructureService, AgentRegistry agentRegistry,
                            AgentUpgradeService agentUpgradeService, SubprocessLogger subprocessLogger, SystemEnvironment systemEnvironment,
-                           PluginManager pluginManager, TaskExtension taskExtension) {
+                           PluginManager pluginManager, PackageAsRepositoryExtension packageAsRepositoryExtension, SCMExtension scmExtension, TaskExtension taskExtension) {
```

Comment:
umm..yes, but I thought having separate args is easier to test interms of mocks etc. Im ok to make it extensions containing all extensions in it.